From 08232c1eababe3c12c94eb1eb272bd363693450e Mon Sep 17 00:00:00 2001
From: Stephan Bergmann <sbergman@redhat.com>
Date: Wed, 9 May 2018 12:15:09 +0200
Subject: [PATCH] Missing template clone() in configmgr dconf mode

...that could cause infinite recursion, e.g. when an erroneously modified
/org.openoffice.ucb.Hierarchy/Entry template (which recursively has children of
the same template type) is later used to instantiate a new set member.

Change-Id: I7b9e55fa1c92979aed98b9f23f4432600afffed4
---
 configmgr/source/dconf.cxx | 1 +
 1 file changed, 1 insertion(+)

diff --git a/configmgr/source/dconf.cxx b/configmgr/source/dconf.cxx
index 22c80841addf..642e37ec0257 100644
--- a/configmgr/source/dconf.cxx
+++ b/configmgr/source/dconf.cxx
@@ -938,6 +938,7 @@ void readDir(
                                 << templ);
                         continue;
                     }
+                    member = member->clone(true);
                     break;
                 default:
                     assert(false); // cannot happen
-- 
2.14.3

